### DISCUSSION TITLE
[Question]Using 'launch_url()' function for page navigation results in original page content updating with URL unchanged, while new page has URL changed but content remains unupdated

### DISCUSSION THREAD

**Discussion by HuaxiaGoodBoy** [flet-dev/flet#4032] [Q&A]
### Question

I want to use the 'launch_url()' function to navigate to a new route, but I found that after using this function, the initial page's URL does not change, but the content becomes the content that the new URL needs to present. On the other hand, the URL address of the new webpage changes, but the content remains the same as the original page content
![屏幕截图 2024-09-22 205107](https://github.com/user-attachments/assets/4c00f4f1-cf5a-42a7-b125-7e72c59b25b7)

### Code sample

```python
import flet as ft

def main(page: ft.Page):
    def go_store(e):
        page.views.append(
            ft.View(
                route="/newpage",
                controls=[
                    ft.AppBar(title=ft.Text("2nd Page"), bgcolor=ft.colors.BLUE),
                ]

            )
        )
        page.launch_url("/newpage")
        page.update()

    page.title = "Routes Example"
    page.add(
        ft.ElevatedButton("Go to Next Page", on_click=go_store),
        ft.AppBar(title=ft.Text("First Page"), bgcolor=ft.colors.SURFACE_VARIANT)
    )

    page.update()


ft.app(main, view=ft.AppView.WEB_BROWSER)
```

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by gual11** [comment]
did you try to set route="/newpage/" ?
